src/App.tsx

```diff
@@ -1,26 +1,238 @@
+        return true;
+    }
+  });
+
+  const addNewTodo = (todoToAdd: Todo): void => {
+    setTempTodo(todoToAdd);
+
+    addTodo(todoToAdd)
+      .then(todoNew => {
+        setTodos(currentTodos => [...currentTodos, todoNew]);
+      })
+      .catch(() => {
+        setErrorType(ErrorType.UnableToAdd);
+      })
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const deleteTodoItem = (todoId: number): void => {
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorType(ErrorType.UnableToDelete);
+      })
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const updateTodoItems = (
+    updateTodoItem: Todo,
+    key: keyof Todo,
+    value: boolean | string,
+  ): Promise<boolean> => {
+    return updateTodo({ ...updateTodoItem, [key]: value })
+      .then(todoUpdated => {
+        setTodos(currentTodos =>
+          currentTodos.map(todo =>
+            todo.id === updateTodoItem.id ? todoUpdated : todo,
+          ),
+        );
+
+        return false;
+      })
+      .catch(() => {
+        setErrorType(ErrorType.UnableToUpdate);
+
+        return true;
+      });
+  };
+
+  const loadedDeleteTodo = (): void => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    setTodoLoading(loadingTodo(completedTodos));
+
+    Promise.allSettled(
+      completedTodos.map(todo => deleteTodo(todo.id).then(() => todo)),
+    )
+      .then(values => {
+        values.forEach(val => {
+          if (val.status === 'rejected') {
+            setErrorType(ErrorType.UnableToDelete);
+          } else {
+            setTodos(currentTodos => {
+              const todoId = val.value as Todo;
+
+              return currentTodos.filter(todo => todo.id !== todoId.id);
+            });
+          }
+        });
+      })
+      .finally(() => setTodoLoading({}));
+  };
+
+  const handleAllCompleted = (): void => {
+    const completedAllTodos = (
+      targetTodos: Todo[],
+      completed: boolean,
+    ): Promise<void> => {
+      return Promise.all(
+        targetTodos.map(todo => updateTodo({ ...todo, completed })),
+      )
+        .then(() => {
+          setTodos(currentTodos =>
+            currentTodos.map(todo =>
+              targetTodos.some(t => t.id === todo.id)
+                ? { ...todo, completed }
+                : todo,
+            ),
+          );
+        })
+        .catch(() => {
+          setErrorType(ErrorType.UnableToUpdate);
+        })
+        .finally(() => {
+          setTodoLoading({});
+        });
+    };
+
+    const activeTodos = todos.filter(todo => !todo.completed);
+
+    if (activeTodos.length) {
+      setTodoLoading(loadingTodo(activeTodos));
+      completedAllTodos(activeTodos, true);
+    } else {
+      setTodoLoading(loadingTodo(todos));
+      completedAllTodos(todos, false);
+    }
+  };
```

Comment:
1) do we need `useCallback` for these function? for which one?
2) use [naming convention](https://dev.to/kristiyan_velkov/react-js-naming-convention-lcg) for all the functions. For all such cases